pkg/server/server.go

```diff
@@ -649,9 +677,20 @@ loop:
 				Callbacks: leaderelection.LeaderCallbacks{
 					OnStartedLeading: func(leaderElectionCtx context.Context) {
 						electionLogger.Info("started leading")
+						if len(s.controllers) == 0 {
+							s.controllerLock.Lock()
+							if err = s.installControllers(ctx, config); err != nil {
+								logger.Error(err, "error in re-registering controllers")
+								return
+							}
+							s.controllerLock.Unlock()
+						}
```

Comment:
Is there a possibility that this and/or `uninstallControllers` is running in multiple Goroutines? I am wondering why we need the mutex. What am I missing? I feel they would run in the same Goroutine.

Reply:
+1, very probably it is one go routine. We could then get rid of the locking?

Reply:
removed locking as suggested